README.md

```diff
@@ -1 +1,156 @@
-# botwo
\ No newline at end of file
+# Boto S3 Router
+
+Interact with S3-compatible services side-by-side with S3, without making any changes to your code. This package provides a boto3-like client that routes requests between S3 clients according to the bucket and the key in the request.
+
+## Why use Boto S3 Router
+
+Consider the following code, downloading an object from one bucket and uploading it to another after some manipulation:
+
+```python
+def calculate(s3_client):
+    obj = s3_client.get_object(Bucket="bucket1", Key="a/b/c/obj")
+    obj2 = do_something_to_obj(obj)
+    s3_client.put_object(Body=obj2, Bucket="bucket2", Key="a/b/c/obj")
+```
+
+Suppose you want to migrate only `bucket2` to an S3-compatible storage like lakeFS or MinIO. 
+Normally, you would have to refactor your code to allow using two S3 clients instead of just one.
+Boto S3 Router allows you to do that without making any changes to the `calculate` function!
+
+## Installation
+
+Boto S3 Router requires Python >= 3.6 to run.
+
+### pip install
+
+If the python package is hosted on a repository, you can install directly using:
+
+```sh
+pip install git+https://github.com/treeverse/botos3router.git
+```
+
+### Setuptools
+
+Install via [Setuptools](http://pypi.python.org/pypi/setuptools).
+
+```sh
+python setup.py install --user
+```
+
+## Basic Usage
+
+```python
+import boto3
+import botos3router as s3r
+
+# Initialize two boto S3 clients according to boto API
+s3_east = boto3.client('s3', region_name='us-east-1', signature_version='v4',)
+s3_west = boto3.client('s3', region_name='us-west-1')
+
+# Defining the rules for the routing between the clients
+profiles = {
+    "s3_west": {
+        "source_bucket_pattern": "bucket",
+        "mapped_bucket_name": "new-bucket",
+        "mapped_prefix": "test/"
+    },
+    "s3_east": {
+        "source_bucket_pattern": "bucket*",
+        "source_key_pattern": "a/*",
+        "mapped_prefix": "test/"
+    }
+}
```

Comment:
Technically this example is valid because the client is chosen by the first profile that matches the bucket name. However, it is really confusing so I'm replacing it

Reply:
Thanks, the new example is much clearer.